File: samples/Go/arithmetic-geometric-mean.go
package main

import (
    "fmt"
    "math"
)

const ε = 1e-14

func agm(a, g float64) float64 {
    for math.Abs(a-g) > math.Abs(a)*ε {
        a, g = (a+g)*.5, math.Sqrt(a*g)
    }
    return a
}

func main() {
    fmt.Println(agm(1, 1/math.Sqrt2))
}

File: samples/Scala/arithmetic-geometric-mean.scala
  def agm(a: Double, g: Double, eps: Double): Double = {
    if (math.abs(a - g) < eps) (a + g) / 2
    else agm((a + g) / 2, math.sqrt(a * g), eps)
  }

  agm(1, math.sqrt(2)/2, 1e-15)
